### PR TITLE
Release/alpha 0.0.1

### PR DESCRIPTION
**Pull request by MariyaSin**

### CODE REVIEW

File: src/UserService.Business/Commands/Credentials/CreateCredentialsCommand.cs [hunk 1]
```diff
@@ -9,126 +9,139 @@
...
+              request.UserId,
+              string.Join('\n', response.Errors));
         }
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, "Something went wrong while we were creating the user credentials");
+      }
+
+      throw new BadRequestException("Something is wrong, please try again later");
```
Comment (BelehovEgor) [inline src/UserService.Business/Commands/Credentials/CreateCredentialsCommand.cs:144]: still throwing exc

File: src/UserService.Business/Commands/User/CreateUserCommand.cs [hunk 2]
```diff
@@ -26,313 +22,316 @@
...
-        private readonly IRequestClient<ISendEmailRequest> _rcSendEmail;
-        private readonly ICreateUserRequestValidator _validator;
-        private readonly IDbUserMapper _mapperUser;
-        private readonly IAccessValidator _accessValidator;
-        private readonly IGeneratePasswordCommand _generatePassword;
-
-        #region private methods
-
-        private void ChangeUserDepartment(Guid departmentId, Guid userId, List<string> errors)
+      string positionErrorMessage = $"Cannot assign position to user. Please try again later.";
```
Comment (BelehovEgor) [inline src/UserService.Business/Commands/User/CreateUserCommand.cs:51]: make const and remove interpolation

File: src/UserService.Business/Commands/User/EditUserCommand.cs [hunk 3]
```diff
@@ -22,337 +19,327 @@
...
-        private readonly IRequestClient<IChangeUserPositionRequest> _rcPosition;
-        private readonly IRequestClient<IChangeUserRoleRequest> _rcRole;
-        private readonly IRequestClient<IChangeUserOfficeRequest> _rcOffice;
-        private readonly IHttpContextAccessor _httpContextAccessor;
-        private readonly IBus _bus;
-
-        #region private method
-
-        private void ChangeUserDepartment(Guid departmentId, Guid userId, List<string> errors)
+      string departmentErrorMessage = $"Cannot assign position to user. Please try again later.";
```
Comment (BelehovEgor) [inline src/UserService.Business/Commands/User/EditUserCommand.cs:46]: make const and remove interpolation

File: src/UserService.Business/Commands/User/EditUserCommand.cs [hunk 4]
```diff
@@ -22,337 +19,327 @@
...
+      if (!(_userRepository.Get(_httpContextAccessor.HttpContext.GetUserId()).IsAdmin ||
+          _accessValidator.HasRights(Rights.AddEditRemoveUsers) ||
+          (userId == requestSenderId
+          && patch.Operations.FirstOrDefault(o => o.path.EndsWith(nameof(EditUserRequest.Rate), StringComparison.OrdinalIgnoreCase)) == null
+          && positionOperation == null
+          && departmentOperation == null
+          && roleOperation == null
+          && officeOperation == null)))
+      {
+        throw new ForbiddenException("Not enough rights.");
```
Comment (BelehovEgor) [inline src/UserService.Business/Commands/User/EditUserCommand.cs:234]: still throwing exc

File: src/UserService.Business/Commands/User/EditUserCommand.cs [hunk 5]
```diff
@@ -22,337 +19,327 @@
...
+        Status = errors.Any()
+              ? OperationResultStatusType.PartialSuccess
+              : OperationResultStatusType.FullSuccess,
+        Body = true,
+        Errors = errors
+      };
+    }
+
+    // TODO fix
+    //private void AddUserSkillsToDbUser(DbUser dbUser, CreateUserRequest request)
```
Comment (BelehovEgor) [inline src/UserService.Business/Commands/User/EditUserCommand.cs:311]: I guess we already can remove it

File: src/UserService.Data/UserCredentialsRepository.cs [hunk 6]
```diff
@@ -13,142 +13,140 @@
...
+      {
+        throw new ArgumentNullException(nameof(userCredentials));
+      }
+
+      _logger.LogInformation(
+        $"Updating user credentials for user '{userCredentials.UserId}'. Request came from IP '{_httpContext.Connection.RemoteIpAddress}'.");
+
+      if (!_provider.UserCredentials.Any(uc => uc.UserId == userCredentials.UserId))
+      {
+        throw new NotFoundException("User credentials was not found.");
```
Comment (BelehovEgor) [inline src/UserService.Data/UserCredentialsRepository.cs:92]: remove exc

**Review by BelehovEgor — APPROVED** [review]

**Review by Cheephot — APPROVED** [review]

File: src/UserService.Business/Commands/Credentials/CreateCredentialsCommand.cs (hunk 1 above)
Comment (MariyaSin) [reply src/UserService.Business/Commands/Credentials/CreateCredentialsCommand.cs:144]: created task 886

File: src/UserService.Business/Commands/User/CreateUserCommand.cs (hunk 2 above)
Comment (MariyaSin) [reply src/UserService.Business/Commands/User/CreateUserCommand.cs:51]: created task 886

File: src/UserService.Business/Commands/User/EditUserCommand.cs (hunk 3 above)
Comment (MariyaSin) [reply src/UserService.Business/Commands/User/EditUserCommand.cs:46]: created task 886

File: src/UserService.Business/Commands/User/EditUserCommand.cs (hunk 4 above)
Comment (MariyaSin) [reply src/UserService.Business/Commands/User/EditUserCommand.cs:234]: created task 886

File: src/UserService.Business/Commands/User/EditUserCommand.cs (hunk 5 above)
Comment (MariyaSin) [reply src/UserService.Business/Commands/User/EditUserCommand.cs:311]: created task 886

File: src/UserService.Data/UserCredentialsRepository.cs (hunk 6 above)
Comment (MariyaSin) [reply src/UserService.Data/UserCredentialsRepository.cs:92]: created task 886

**Pull request merged**